libsolidity/ast/Types.cpp

```diff
@@ -1503,18 +1388,32 @@ u256 BoolType::literalValue(Literal const* _literal) const
 TypeResult BoolType::unaryOperatorResult(Token _operator) const
 {
 	if (_operator == Token::Delete)
-		return TypeResult(make_shared<TupleType>());
-	return (_operator == Token::Not) ? shared_from_this() : TypePointer();
+		return TypeProvider::emptyTupleType(); // TODO: Lag of understanding.
```

Comment:
```suggestion
		return TypeProvider::emptyTupleType();
```

Reply:
`delete` returns the empty tuple because the type of the operator result is "nothing", i.e. you cannot use it inside any other expression.